raiden-ts/src/transport/epics/webrtc.ts

```diff
@@ -81,37 +108,36 @@ async function getMatrixIceServers(matrix: MatrixClient): Promise<RTCIceServer[]
 }
 
 // creates a filter function which filters valid MatrixEvents
-function filterMatrixVoipEvents<
-  T extends 'm.call.invite' | 'm.call.answer' | 'm.call.candidates' | 'm.call.hangup'
->(type: T, sender: string, callId: string, httpTimeout?: number) {
-  type ContentKey = T extends 'm.call.invite'
-    ? 'offer'
-    : T extends 'm.call.answer'
-    ? 'answer'
-    : T extends 'm.call.candidates'
-    ? 'candidates'
-    : never;
-  const contentKey = (type === 'm.call.invite'
-    ? 'offer'
-    : type === 'm.call.answer'
-    ? 'answer'
-    : type === 'm.call.candidates'
-    ? 'candidates'
-    : undefined) as ContentKey | undefined;
-  return (
-    // FIXME: remove any when MatrixEvent type exposes getAge & getContent methods
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    event: any,
-  ): event is MatrixEvent & {
-    getType: () => T;
+function filterMatrixVoipEvents<T extends 'offer' | 'answer' | 'candidates' | 'hangup'>(
```

Comment:
Enum would make it harder to express the unions as above. I'll check how to do it, but would say the literals should be enough, and since this is internal to the protocol/module, it isn't very concerning that it isn't an enum.

Reply:
Ok, managed to get it in a enum even with the union type. Thanks for the suggestion.

Reply:
Cool. Thanks for the invested time. Really like it. :tada: